transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java

```diff
@@ -120,10 +120,31 @@ protected void run() {
 
         // Lets add the eventfd related events before starting to do any real work.
         submissionQueue.addPollIn(eventfd.intValue());
-        submissionQueue.submit();
 
         for (;;) {
             logger.trace("Run IOUringEventLoop {}", this.toString());
+            // avoid blocking for as long as possible - loop until two consecutive "empty" results
+            boolean maybeMoreTasks = true;
+            boolean maybeMoreIo;
+            do {
+                maybeMoreIo = completionQueue.process(this) != 0;
+                if (!maybeMoreIo & !maybeMoreTasks) {
```

Comment:
should be `&&`?

Reply:
@nitsanw this is probably more in your wheelhouse but my understanding was that non-short-circuit variants of boolean ops are preferable when the operands are trivial (eliminates a branch I think)? I've seen it used in other places but agree it looks odd.